omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptReferenceTermSearchHandler1_9.java

```diff
@@ -62,41 +69,101 @@ public PageableResult search(RequestContext context) throws ResponseException {
 		String source = context.getParameter("source");
 		String code = context.getParameter("code");
 		String name = context.getParameter("name");
-		
+		String searchType = context.getParameter("searchType");
+
+		if(searchType!=null&&!SEARCH_TYPE_EQUALS.equals(searchType)&&!SEARCH_TYPE_STARTS_WITH.equals(searchType)){
+			throw new InvalidSearchException("Invalid searchType parameter");
+		}
+
 		ConceptSource conceptSource = conceptService.getConceptSourceByUuid(source);
-		
+
 		if (conceptSource == null) {
 			conceptSource = conceptService.getConceptSourceByName(source);
 		}
-		
+
 		if (conceptSource == null) {
 			return new EmptySearchResult();
 		}
-		
+
 		if (code != null || name != null) {
-			ConceptReferenceTerm term = null;
-			if (code != null) {
-				term = conceptService.getConceptReferenceTermByCode(code, conceptSource);
-				if (name != null && !name.equals(term.getName())) {
-					term = null;
-				}
+			if(SEARCH_TYPE_STARTS_WITH.equals(searchType)){
+				return searchStartsWith(context, code, name, conceptSource);
 			} else {
-				term = conceptService.getConceptReferenceTermByName(name, conceptSource);
-			}
-			
-			if (term == null || (term.isRetired() && !context.getIncludeAll())) {
-				return new EmptySearchResult();
-			} else {
-				return new AlreadyPaged<ConceptReferenceTerm>(context, Arrays.asList(term), false);
+				return searchEquals(context, code, name, conceptSource);
 			}
 		} else {
 			List<ConceptReferenceTerm> terms = conceptService.getConceptReferenceTerms(null, conceptSource,
-			    context.getStartIndex(), context.getLimit(), context.getIncludeAll());
+					context.getStartIndex(), context.getLimit(), context.getIncludeAll());
 			int count = conceptService.getCountOfConceptReferenceTerms(null, conceptSource, context.getIncludeAll());
 			boolean hasMore = count > (context.getStartIndex() + context.getLimit());
-			
+
 			return new AlreadyPaged<ConceptReferenceTerm>(context, terms, hasMore);
 		}
 	}
-	
+

```

Comment:
Why are we doing something in rest that is not supported in the Java API? I personally don't agree with doing this